packages/components/src/disclosure/types.tsx

```diff
@@ -0,0 +1,10 @@
+export type DisclosureContentProps = {
+	/**
+	 * If set to `true` the content will be shown, otherwise it's hidden.
+	 */
+	visible?: boolean;
```

Comment:
I know we're inheriting this from the original API, but I find it confusing. Ariakit's `DisclosureProvider` supports a `open` prop (on top of [a few more useful ones](https://ariakit.org/reference/disclosure-provider#optional-props)), and if I want to compose my own disclosure UI by using the provider and the disclosure components, it feels odd that I have to pass `visible` once again and that it doesn't pick it up from the provided context.

I think now that we're doing this migration it is a good time to revisit the API and deprecate this prop in favor of using the provider. It just feels like we're adding an unnecessary layer on top of what already exists as provided by the underlying library.

Reply:
Retracting this comment as this decision was clarified in https://github.com/WordPress/gutenberg/pull/55639#issuecomment-1794316403.